### PR TITLE
Added probe to SDS

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Added readiness probes to any pod with an SDS, this includes the gloo deployment and gateway proxy deployment.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5557

### CODE REVIEW

File: changelog/v1.12.0-beta3/sds-readiness-probe.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
```
Comment (saiskee) [inline changelog/v1.12.0-beta3/sds-readiness-probe.yaml:2]: `NEW_FEATURE`?

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml [hunk 2]
```diff
@@ -281,6 +281,12 @@ spec:
       - name: sds
         image: {{ template "gloo.image" $sdsImage }}
         imagePullPolicy: {{ $sdsImage.pullPolicy }}
+        readinessProbe:
+          tcpSocket:
+            port: {{ .Values.gatewayProxies.gatewayProxy.XdsServicePort }}
```
Comment (saiskee) [inline install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:286]:
```suggestion
            port: {{ $spec.xdsServicePort }}
```

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml [hunk 3]
```diff
@@ -281,6 +281,12 @@ spec:
       - name: sds
         image: {{ template "gloo.image" $sdsImage }}
         imagePullPolicy: {{ $sdsImage.pullPolicy }}
+        readinessProbe:
+          tcpSocket:
+            port: {{ $spec.xdsServicePort }}
```
Comment (saiskee) [inline install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:286]: let's hardcode this to `8234` based on the port value [here](https://github.com/solo-io/gloo/blob/46aaa649da717f88c878742e21d7a7d2611c8e9d/projects/sds/cmd/main.go#L25)

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml [hunk 4]
```diff
@@ -290,6 +290,12 @@ spec:
 {{- if $global.glooMtls.sdsResources }}
         resources:
 {{ toYaml $global.glooMtls.sdsResources | indent 10}}
+        readinessProbe:
+          tcpSocket:
+            port: {{ $spec.podTemplate.httpPort }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:295]: i think the http port here is for the envoy container https://github.com/solo-io/gloo/blob/e540ed56fa5481da47740d3e1b900a12ef812c83/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml#L196 not the sds container https://github.com/solo-io/gloo/blob/e540ed56fa5481da47740d3e1b900a12ef812c83/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml#L287

we should probably be using readiness on the sds container port instead: https://github.com/solo-io/gloo/blob/e540ed56fa5481da47740d3e1b900a12ef812c83/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml#L330-L332

**Review by kdorosh — CHANGES_REQUESTED** [review]
overall looks good, great job sorting out our complex helm charts!

just looking for one change to a probe; also lets update our helm tests to ensure these render correctly

File: install/helm/gloo/templates/7-gateway-proxy-deployment.yaml (hunk 4 above)
Comment (kdorosh) [reply install/helm/gloo/templates/7-gateway-proxy-deployment.yaml:295]: I think this isn't rendering, it should be moved out of the `{{ toYaml $global.glooMtls.sdsResources | indent 10}}` check

**Pull request closed without merging**